modules/dxp/apps/sharepoint-soap/sharepoint-soap-repository/src/main/java/com/liferay/sharepoint/soap/repository/connector/internal/util/test/SharepointConnectionTestUtil.java

```diff
@@ -80,9 +82,13 @@ private static String _getSharepointVMHostname() throws Exception {
 		try (BufferedReader bufferedReader = new BufferedReader(
 				new InputStreamReader(httpURLConnection.getInputStream()))) {
 
-			Stream<String> payloadStream = bufferedReader.lines();
+			List<String> data = new ArrayList<>();
 
-			String payload = payloadStream.collect(Collectors.joining());
+			while (bufferedReader.ready()) {
```

Comment:
@jeffwu0724 Please use `while ((line = bufferedReader.readLine()) != null)` here. `ready()` only returns true when there is data in the buffer to be read. When reading from a network socket, it's common that the socket/inputstream is not closed, but there's no data to read -- the other end doesn't send any. In this case, `ready()` will return `false`.

Reply:
@jeffwu0724 Also, please directly use a StringBundler.